Cubical/Homotopy/WhiteheadsLemma.agda

```diff
@@ -0,0 +1,447 @@
+     ∘ (fst g)
+     ∘ (Iso.fun (pathToIso (λ i → fst (p i))))
+     ≡ (fst f)
+  ∙PathP→Square {A = A} {B = B} {C = C} {D = D} =
+    J ( λ C' p →
+                  (q : B ≡ D) (f : A →∙ B) (g : C' →∙ D)
+               → PathP (λ i → (p i) →∙ (q i)) f g
+               → (Iso.inv (pathToIso (λ i → fst (q i))))
+                 ∘ (fst g)
+                 ∘ (Iso.fun (pathToIso (λ i → fst (p i))))
+                 ≡ (fst f))
+      ( J ( λ D' q →
+                      (f : A →∙ B) (g : A →∙ D')
+                   → PathP (λ i → A →∙ (q i)) f g
+                   → (Iso.inv (pathToIso (λ i → fst (q i))))
+                    ∘ (fst g)
+                    ∘ (Iso.fun (pathToIso (λ i → (fst A))))
+                    ≡ (fst f))
+          ( λ f g r →
+            funExt ( λ a → transportRefl (fst g (transport refl a))
+                          ∙ cong (fst g) (transportRefl a)
+                          ∙ λ i → (fst (r (~ i))) a)))
+
+  doubleCompPath-filler-refl : {A : Type ℓ} {a b : A} (p : a ≡ b)
+    → p ≡ refl ∙∙ p ∙∙ refl
+  doubleCompPath-filler-refl p = doubleCompPath-filler refl p refl
+
+  isContr→isContr→isEquiv : {A B : Type ℓ} (f : A → B)
+    → isContr A → isContr B → isEquiv f
+  equiv-proof (isContr→isContr→isEquiv f (a , hA) (b , hB)) b' =
+    ( a , isContr→isProp (b , hB) (f a) b') ,
+    ( λ (a' : fiber f b') → ΣPathP
+              ( isContr→isProp (a , hA) a (fst a')
+              , ( toPathP
+                  ( isProp→isSet
+                    ( isContr→isProp (b , hB))
+                    ( f _) b'
+                    ( transport (λ i → f _ ≡ b') _) (snd a')))))
+
+  isPointedTarget→isEquiv→isEquiv : {A B : Type ℓ} (f : A → B)
+    → (B → isEquiv f) → isEquiv f
+  equiv-proof (isPointedTarget→isEquiv→isEquiv f hf) =
+    λ y → equiv-proof (hf y) y
+
+  squareWithEquivs→Equiv : {A B C D : Type ℓ}
+    (f : A → B) (e1 : C → D) (e2 : A → C) (e3 : D → B)
+    → isEquiv e1 → isEquiv e2 → isEquiv e3 → e3 ∘ e1 ∘ e2 ≡ f
+    → isEquiv f
+  squareWithEquivs→Equiv f e1 e2 e3 e1Eq e2Eq e3Eq p =
+    transport
+    ( λ i → isEquiv (p i))
+    ( snd (compEquiv
+           ( e2 , e2Eq)
+           ( (e3 ∘ e1) , (snd (compEquiv (e1 , e1Eq) (e3 , e3Eq))))))
+-- end of the long private block of helper lemmas
+
+-- some useful paths between different maps
+
+{-
+
+a ≡ b ------> f a ≡ f b
+  |               ^
+  |               |
+  |               |
+  v               |
+a ≡ a ------> f a ≡ f a
+
+-}
+ΩCongSquare : {A B : Type ℓ} {a b : A} (f : A → B) (p : a ≡ b)
+  → (λ q → q ∙ (cong f p))
+   ∘ fst (Ω→ (f , refl))
+   ∘ (λ q → q ∙ sym p)
+   ≡ cong f
+ΩCongSquare f =
+  J ( λ b p → (λ q → q ∙ (cong f p))
+             ∘ fst (Ω→ (f , refl))
+             ∘ (λ q → q ∙ sym p)
+             ≡ cong f)
+    ( funExt (λ x → sym (rUnit _)
+                   ∙ sym (doubleCompPath-filler-refl _)
+                   ∙ cong (cong f) (sym (rUnit _))))
+
+private
+  setMapFunctorial : {A B C : Type ℓ} (f : A → B) (g : B → C)
```

Comment:
Oh wow, we really should have this one lol... Could you put it in HITs.SetTruncation.Properties? And maybe just call it `mapFunctorial` to match the naming there better.